src/lib/db/environment-store.ts

```diff
@@ -92,12 +106,44 @@ export default class EnvironmentStore implements IEnvironmentStore {
         return mapRow(row);
     }
 
-    async upsert(env: IEnvironment): Promise<IEnvironment> {
+    async updateProperty(
+        id: string,
+        field: string,
+        value: string | number,
+    ): Promise<void> {
         await this.db<IEnvironmentsTable>(TABLE)
-            .insert(mapInput(env))
-            .onConflict('name')
-            .merge();
-        return env;
+            .update({
+                [field]: value,
+            })
+            .where({ name: id, protected: false });
+    }
+
+    async updateSortOrder(id: string, value: number): Promise<void> {
+        await this.db<IEnvironmentsTable>(TABLE)
+            .update({
+                sort_order: value,
+            })
+            .where({ name: id });
+    }
+
+    async update(
+        env: Pick<IEnvironment, 'displayName' | 'type' | 'protected'>,
```

Comment:
Why do we care about 'protected', I thought some of the point here was that it's only displayName, type and enabled we're allowed to change?

Reply:
Actually we only care about displayName and type when updating, but we override protected in the service.